blog/content/post/2024-02-07-cross2024_ftpm_tee_ta.md

```diff
@@ -0,0 +1,112 @@
+pivotal role by safeguarding cryptographic keys and other sensitive data used
+in authentication processes. Traditionally, TPMs are external hardware
+components, adding a layer of physical security but also increasing the overall
+cost of device manufacturing. However, in the quest to balance
+cost-effectiveness with security, one viable solution that emerges is the
+integration of firmware-based TPMs (fTPMs) within the CPU itself. This approach
+presents an innovative alternative for scenarios where traditional hardware
+TPMs may not be feasible or cost-effective but also come with fallbacks.
+Fundamentally it requires the establishment of a Trusted Execution Environment
+(TEE) to securely manage cryptographic operations. Fortunately, ARM's TrustZone
+technology provides a robust framework to implement TEE on embedded ARM
+devices, blending efficiency with security.
+
+## Arm TrustZone
+
+For Arm Cortex-A, Arm TrustZone creates two distinct memory "worlds": a Normal
+World for the Operating System (referred to as Rich OS in documentation) and
+a Secure World for implementing the Trusted Execution Environment (TEE).
+The transition between these worlds is managed by the Secure Monitor, operating at
+a higher exception level (EL3), ensuring secure memory regions are exclusively
+accessible from the Secure World. This mechanism supports running fTPM in the
+Secure World, enabling secure syscalls from user space. Secrets stored in fTPM
+are secure as long as the Secure Monitor is not compromised.
+
+![Cortex-A TrustZone Exception Levels](/img/TEE_ARM_Cortex-a_exception_levels.svg)
+![Cortex-A TrustZone](/img/TEE_ARM_Cortex-a.svg)
+
+Arm TrustZone also exists for the Cortex-m series but adopts a simpler and more
+hardware-focused approach relying on hardware mechanisms to manage the CPU
+state via interrupts.
+
+fTPM requires a non-trivial amount of computational
+resources and memory, which might be scarce in the environments where Cortex-M
+processors are typically used. Implementing fTPM could therefore be impractical
+due to the limited resources available on these devices. It's also rare for the
+Cortex-M devices demand the complex security functionalities that fTPM
+provides.
+
+![Cortex-M TrustZone](/img/TEE_ARM_Cortex-m.svg)
+
+## Fallbacks and Security Concerns
+
+_The best-protected systems have dedicated hardware security measures included
+from the beginning of their design process, starting with the specification
+for the processor core and the SoC infrastructure._
+
+As fTPM is implemented purely in software the huge advantage is that it can be
+added to already provisioned devices via an update. The caveat is that while
+this can improve the security of such devices there are hardware security
+concerns that the device should fulfill from the beginning. For example
+the [official OP-TEE Documentation specifies the Raspberry Pi 3 platform as not
+suitable for a secure implementation of Trusted Execution Environment](
```

Comment:
@tym2k1 does this link renders correctly?

Reply:
There is a small hiccup with displaying the word "official"
![image](https://github.com/3mdeb/news-and-ideas/assets/32579616/da11da16-30dd-4822-b02e-429c0dee846e)

Reply:
It looks ok, I am closing this conversation